utils/db_process.py

```diff
@@ -0,0 +1,93 @@
+import mysql.connector
+from mysql.connector import pooling
+import os
+from dotenv import load_dotenv
+from typing import Union, Tuple, Dict, List
+from typing import Final
+import datetime
+load_dotenv()
+
+pool_config = {
+    "pool_name": "pool",
+    "pool_size": 5,
+    "host": os.getenv("DB_HOST"),
+    "user": os.getenv("DB_USER"),
+    "password": os.getenv("DB_PASSWORD"),
+    "database": os.getenv("DB_DATABASE"),
+}
+pool_config: Final
+
+
+cnx_pool = pooling.MySQLConnectionPool(**pool_config)
+
+
+def get_all_result(sql, param: Union[Tuple, None] = None) -> Dict:
+    try:
+        connection = cnx_pool.get_connection()
+        cursor = connection.cursor()
+
+        if param is None:
+            cursor.execute(sql)
+        else:
+            cursor.execute(sql, param)
+        result = cursor.fetchall()
+        
+        cursor.close()
+        connection.close()
+        column_names = [desc[0] for desc in cursor.description]
+        result = [dict(zip(column_names, row)) for row in result]
+        return result
+
+    except mysql.connector.Error as e:
+        print(e)
+        cursor.close()
+        connection.close()
+        return False
+
+def execute_query(sql, param: Union[Tuple, None] = None) -> bool:
+    try:
+        connection = cnx_pool.get_connection()
+        cursor = connection.cursor()
+
+        if param is None:
+            count = cursor.execute(sql)
+        else:
+            count = cursor.execute(sql, param)
+        if count == 0:
+            cursor.close()
+            connection.close()
+            return False
+        connection.commit()
+        cursor.close()
+        connection.close()
+        return True
+
+    except mysql.connector.Error as e:
+        print(e)
+        cursor.close()
+        connection.rollback()
+        connection.close()
+        return False
+    
+def dict_to_sqltext(_dict: Dict, exclude_col: List[str] = []) -> Tuple[str, tuple]:
```

Comment:
> If a mutable object (like a list, dictionary, or any instance of a user-defined class) is used as a default argument in a Python function, it can lead to unexpected behavior. That's because function argument defaults are evaluated only once at function definition time, not each time the function is called.
Here's an example to demonstrate this:
```python
def append_to(num, target=[]):
    target.append(num)
    return target

print(append_to(1))  # outputs: [1]
print(append_to(2))  # outputs: [1, 2], not [2] as you might expect
```

Reply:
```python
def dict_to_sqltext(_dict: Dict, exclude_col: List[str] = None) -> Tuple[str, tuple]:
    if exclude_col is None:
        exclude_col = []
```
would do it